app/frontend/wallet/shelley/shelley-trezor-crypto-provider.ts

```diff
@@ -79,9 +78,9 @@ const CardanoTrezorCryptoProvider = async ({network, config}) => {
     certificatePointer?: CardanoCertificatePointer
   }
 
-  async function displayAddressForPath(absDerivationPath, stakingPath?) {
+  async function displayAddressForPath(absDerivationPath: BIP32Path, stakingPath?: BIP32Path) {
     const addressParameters: CardanoAddressParameters = {
-      addressType: 0,
+      addressType: 0, // TODO: get from cardano-crypto
```

Comment:
looks like a good opportunity to introduce AddressType enum - I'd not rely on cardano-crypto as it does not have proper typescript definitions, so I would not be afraid to declare our own enum here

Reply:
Sure I ll add the enum, but the comment meant that we should get the addressType out of the address it self using `cardano-crypto` and not hardcode it here hoping it will always be a Base address.

Reply:
I see, then this should perhaps be a general comment for the function that it should be extended to accept the expected address type as that's not being passed to it currently - in theory, you can figure it out from the presence/absence of the staking/spending path but seems rather fragile

Reply:
I ll comment it better :+1: